net/url_request/url_fetcher.h

```diff
@@ -381,6 +389,15 @@ class NET_EXPORT URLFetcher {
   // set to store the response as a string.
   virtual bool GetResponseAsString(std::string* out_response_string) const = 0;
 
+  #if defined(IN_MEMORY_UPDATES)
+  // Get the resonse as a string when the response is expected to be large and
+  // the data should be "moved" and not copied. Should only be used in
+  // conjunction with SaveResponseToLargeString() and returns false if the
+  // fetcher was not set to store the response as a "large" string.
+  virtual bool GetResponseAsLargeString(std::string* out_response_string)
```

Comment:
Same as above. The API additions seem unnecessary, as `URLFetcherResponseWriter` is already provided in the API as a customization hook I don't know there's a justified big enough difference between this API function and `GetResponseAsString` getting it from custom ResponseWriter that pre-allocates.

Overall, i don't think there's a need to change a single line of code in `net/url_request` for this functionality.

Reply:
How do you envision GetResponseAsString() "getting it from a custom ResponseWriter that pre-allocates" without "chang[ing] a single line of code in `net/url_request`"?

If you look at the call stack, GetResponseAsString() ultimately uses `URLFetcherCore::GetResponseAsString()` to call the `data()` method from a `URLFetcherStringWriter`, which does not preallocate: https://github.com/youtube/cobalt/blob/main/net/url_request/url_fetcher_core.cc#L427.

Reply:
Have a look at how streaming fetch was implemented in Cobalt: https://github.com/youtube/cobalt/blob/main/cobalt/xhr/xml_http_request.cc#L1014

Reply:
Thanks - will do - sounds like you're ok with the PR going in as-is, though, so resolving this.